### PR TITLE
Add builder dev report for 2017-07-10

### PR DESCRIPTION
**Pull request by tonistiigi**
@vieux @thaJeztah 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

File: reports/builder/2017-07-10.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+### BuildKit
+
+[Repo](https://github.com/moby/buildkit)
+[Proposal](https://github.com/moby/moby/issues/32925)
+
+Many new features have been added since the last report.
+
+The build definition solver was updated to detect the identical parts of the graph sent by different clients and synchronize their processing. This is important when multiple targets of the same project are built on the same time and removes any duplication of work.
+
+Running build jobs now has support for graceful canceling and clear error reporting in case some build steps fail or are canceled. Bugs that may have left state dir in an inconsistent state of server shutdown were fixes.
```
Comment (AkihiroSuda) [inline reports/builder/2017-07-10.md]: fixes -> fixed

File: reports/builder/2017-07-10.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+The first version of instruction cache was implemented. This caching scheme has many benefits compared to the current `docker build` caching as it doesn't require all data to be locally available to determine the cache match. The interface for the cache implementation is much simpler and could be implemented remotely as it only needs to store the cache keys and doesn't need to understand or compare their values. Content-based caching will be implemented on top of this work later.
+
+Separate source implementation for git repositories is currently in review. Using this source for accessing source code in git repositories has many performance and caching advantages. All the build jobs using the same git remote will use a shared local repository where updates will be pulled. All the nodes based on a git source will be cached using the commit ID of the current checkout.
```
Comment (AkihiroSuda) [inline reports/builder/2017-07-10.md]: in review -> merged

File: reports/builder/2017-07-10.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+
+### BuildKit
+
+[Repo](https://github.com/moby/buildkit)
+[Proposal](https://github.com/moby/moby/issues/32925)
+
+Many new features have been added since the last report.
+
+The build definition solver was updated to detect the identical parts of the graph sent by different clients and synchronize their processing. This is important when multiple targets of the same project are built on the same time and removes any duplication of work.
```
Comment (thaJeztah) [inline reports/builder/2017-07-10.md]: s/on the same time/at the same time/

File: reports/builder/2017-07-10.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+### Typed Dockerfile parsing
+
+[PR](https://github.com/moby/moby/pull/33492)
+
+The PR is in code review and waiting for feedback. Hopefully ready to me merged this week.
+
+### Quality: Dependency interface switch
+
+No updates for this week. Metadata commands need to be updated but it probably easier to do it after https://github.com/moby/moby/pull/33492 has been merged.
```
Comment (thaJeztah) [inline reports/builder/2017-07-10.md]: s/but it probably easier/but it is probably easier/

**Review by thaJeztah — CHANGES_REQUESTED** [review]
minor nits, but LGTM

File: reports/builder/2017-07-10.md (hunk 2 above)
Comment (tonistiigi) [inline reports/builder/2017-07-10.md]: I wrote this on Monday. Sorry for posting late.

**Review by thaJeztah — APPROVED** [review]
LGTM

ping @vieux

File: reports/builder/2017-07-10.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Separate source implementation for git repositories is currently in review. Using this source for accessing source code in git repositories has many performance and caching advantages. All the build jobs using the same git remote will use a shared local repository where updates will be pulled. All the nodes based on a git source will be cached using the commit ID of the current checkout.
+
+Next areas to be worked on will be implementing first exporters for getting access to the build artifacts and porting over the client session/incremental-send feature from `17.07-ce`.
+
+### Typed Dockerfile parsing
+
+[PR](https://github.com/moby/moby/pull/33492)
+
+The PR is in code review and waiting for feedback. Hopefully ready to me merged this week.
```
Comment (cpuguy83) [inline reports/builder/2017-07-10.md]: "be merged"

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: reports/builder/2017-07-10.md (hunk 5 above)
Comment (tonistiigi) [inline reports/builder/2017-07-10.md]: updated

**Comment by thaJeztah** [comment]
I see @cpuguy83's comment was addressed; I'll merge

**Pull request merged**